Editor/Utility/CinemachineSceneTools.cs

```diff
@@ -0,0 +1,564 @@
+        public const float lineThickness = 4f;
+
+        static GUIStyle s_LabelStyle = new GUIStyle 
+        { 
+            normal =
+            {
+                background = AssetDatabase.LoadAssetAtPath<Texture2D>(ScriptableObjectUtility.
+                    CinemachineRealativeInstallPath + "/Editor/EditorResources/SceneToolsLabelBackground.png"),
+                textColor = Handles.selectedColor,
+            },
+            fontStyle = FontStyle.Bold,
+            padding = new RectOffset(5, 0, 5, 0)
+        };
+        
+
+        public static float SliderHandleDelta(Vector3 newPos, Vector3 oldPos, Vector3 forward)
+        {
+            var delta = newPos - oldPos;
+            var sameDirection = Vector3.Dot(delta.normalized, forward) > 0;
+            return (sameDirection ? 1f : -1f) * delta.magnitude;
+        }
+
+        /// <summary>
+        /// Calculate delta and discard imprecision.
+        /// </summary>
+        public static Vector3 PositionHandleDelta(Quaternion rot, Vector3 newPos, Vector3 oldPos)
+        {
+            var delta =
+                Quaternion.Inverse(rot) * (newPos - oldPos);
+            delta = new Vector3(
+                Mathf.Abs(delta.x) < UnityVectorExtensions.Epsilon ? 0 : delta.x,
+                Mathf.Abs(delta.y) < UnityVectorExtensions.Epsilon ? 0 : delta.y,
+                Mathf.Abs(delta.z) < UnityVectorExtensions.Epsilon ? 0 : delta.z);
+            return delta;
+        }
+        
+        public static void DrawLabel(Vector3 position, string text)
+        {
+            var labelOffset = HandleUtility.GetHandleSize(position) / 5f;
+            Handles.Label(position + new Vector3(0, -labelOffset, 0), text, s_LabelStyle);
+        }
+
+        static int s_ScaleSliderHash = "ScaleSliderHash".GetHashCode();
+        public static void FovToolHandle(CinemachineVirtualCameraBase vcam, ref LensSettings lens, 
+            bool isLensHorizontal, ref float fov, ref bool soloSetByTools)
+        {
+            var camPos = vcam.State.FinalPosition;
+            var camRot = vcam.State.FinalOrientation;
+            var camForward = camRot * Vector3.forward;
+                
+            EditorGUI.BeginChangeCheck();
+            var fovHandleId = GUIUtility.GetControlID(s_ScaleSliderHash, FocusType.Passive) + 1; // TODO: KGB workaround until id is exposed
+
+            var newFov = Handles.ScaleSlider(
+                fov, 
+                camPos, camForward, camRot, HandleUtility.GetHandleSize(camPos), 0.1f);
+            if (EditorGUI.EndChangeCheck())
+            {
+                Undo.RecordObject(vcam, "Changed FOV using handle in scene view.");
+
+                if (lens.Orthographic)
+                {
+                    lens.OrthographicSize += (fov - newFov);
+                }
+                else
+                {
+                    lens.FieldOfView += (fov - newFov);
+                }
+                lens.Validate();
```

Comment:
Does this get called automatically thanks to Undo?  Same question for other calls to Validate() elsewhere in this file